src/transformers/models/speecht5/modeling_speecht5.py

```diff
@@ -2628,6 +2713,54 @@ def forward(
             encoder_attentions=outputs.encoder_attentions,
         )
 
+    def _compute_loss(
+        self,
+        attention_mask: torch.Tensor,
+        outputs_before_postnet: torch.FloatTensor,
+        outputs_after_postnet: torch.FloatTensor,
+        logits: torch.FloatTensor,
+        labels: torch.FloatTensor,
+        cross_attentions: Optional[torch.FloatTensor] = None,
+    ):
+        padding_mask = labels != -100.0
+
+        # mask out the padded portions
+        labels = labels.masked_select(padding_mask)
+        outputs_before_postnet = outputs_before_postnet.masked_select(padding_mask)
+        outputs_after_postnet = outputs_after_postnet.masked_select(padding_mask)
+
+        # spectrogram loss
+        l1_criterion = L1Loss()
+        l1_loss = l1_criterion(outputs_after_postnet, labels) + l1_criterion(outputs_before_postnet, labels)
+
+        # construct stop labels from the padding mask
+        masks = padding_mask[:, :, 0]
+        stop_labels = torch.cat([~masks * 1.0, torch.ones(masks.size(0), 1).to(masks.device)], dim=1)
+        stop_labels = stop_labels[:, 1:].masked_select(masks)
+        logits = logits.masked_select(masks)
+
+        # stop token loss
+        bce_criterion = BCEWithLogitsLoss(pos_weight=torch.tensor(5.0))
+        bce_loss = bce_criterion(logits, stop_labels)
+
+        loss = l1_loss + bce_loss
+
+        # guided attention loss
+        if self.config.use_guided_attention_loss:
+            attn_criterion = SpeechT5GuidedMultiheadAttentionLoss(
+                sigma=self.config.guided_attention_loss_sigma,
+                scale=self.config.guided_attention_loss_scale,
+            )
+            attn = torch.cat([x[:, : self.config.guided_attention_loss_num_heads] for x in cross_attentions], dim=1)
+            input_masks = attention_mask == 1
```

Comment:
`attn_criterion` would work fine if we passed `attention_mask` as a LongTensor of 0/1 values no?

Reply:
I think it's clearer by explicitly making it a True/False tensor, since that's what it will end up being anyway (to do a masked select).